src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataSourceController.java

```diff
@@ -107,13 +109,28 @@ public BaseDataSourceController(GenericConversionService conversionService,
             DS dataSource = convertDTOToDataSource(commonDataSourceDTO);
             dataSource.setId(dataSourceId);
             dataSource.setDataNode(exist.getDataNode());
+            dataSource.setPublished(true);
             dataSource = dataSourceService.update(dataSource);
             result = new JsonResult<>(NO_ERROR);
             result.setResult(convertDataSourceToDTO(dataSource));
         }
         return result;
     }
 
+    @RequestMapping(value = "/api/v1/data-sources/{id}/from-node", method = RequestMethod.PUT)
+    public JsonResult<DTO> updateFieldsDefinedAtNode(
+            @PathVariable("id") Long dataSourceId,
+            @RequestBody DTO commonDataSourceDTO
+    ) throws IllegalAccessException, IOException, NoSuchFieldException, SolrServerException, ValidationException {
+
+        JsonResult<DTO> result = new JsonResult<>(NO_ERROR);
+        DS updating = convertDTOToDataSource(commonDataSourceDTO);
+        updating.setId(dataSourceId);
+        updating = dataSourceService.update(updating);
```

Comment:
Where is permission check for DNs requests only?

Reply:
there is EDIT_DATASOURCE permission check (update method of BaseDataSourceServiceImpl)